src/ZeroLog/LogEvent.cs

```diff
@@ -16,19 +16,24 @@ internal unsafe partial class LogEvent : IInternalLogEvent
         private const int _maxArgCapacity = byte.MaxValue;
         private string[] _strings;
         private IntPtr[] _argPointers;
+        private IntPtr[] _keyValuePointers;
         private Log _log;
         private LogEventArgumentExhaustionStrategy _argumentExhaustionStrategy;
 
         protected readonly byte* _startOfBuffer;
         protected readonly byte* _endOfBuffer;
         protected byte* _dataPointer;
         private byte _argCount;
+        private byte _keyValueCount;
```

Comment:
Same as above - just use a single `List<IntPtr>` in the appender thread instead of `_keyValuePointers` and `_keyValueCount`. You will even be able to preallocate it to a large enough size.

Reply:
Done, thanks!